bookkeeper-server/src/test/java/org/apache/bookkeeper/test/BookKeeperClusterTestCase.java

```diff
@@ -480,16 +485,14 @@ public void restartBookie(BookieSocketAddress addr) throws Exception {
         int toRemoveIndex = 0;
         for (BookieServer server : bs) {
             if (server.getLocalAddress().equals(addr)) {
-                server.shutdown();
                 toRemove = server;
                 break;
             }
             ++toRemoveIndex;
         }
         if (toRemove != null) {
-            stopAutoRecoveryService(toRemove);
```

Comment:
This line seems to related to the issue

Reply:
Sorry for typo, it does not seem to be related to the issue

Reply:
stopAutoRecoveryService gets called as part of of killBookie.  I refactored it a little to make the bs/bsLoggers state maintenance slightly less error prone.